### ISSUE TITLE
Tracer is issuing excessive amounts of stat() syscalls

### ISSUE THREAD

**Issue by whitequark** [m-labs/nmigen#170]
For unknown reasons, the strace of any nMigen synthesis run looks like this:
```
stat("/home/whitequark/Projects/nmigen/nmigen/tracer.py", {st_mode=S_IFREG|0644, st_size=1883, ...}) = 0
stat("/home/whitequark/Projects/nmigen/nmigen/hdl/ast.py", {st_mode=S_IFREG|0644, st_size=47434, ...}) = 0
stat("/home/whitequark/Projects/nmigen/nmigen/hdl/xfrm.py", {st_mode=S_IFREG|0644, st_size=22180, ...}) = 0
```
And this goes on for several tens of thousands of lines. This is likely why nMigen elaboration is so much slower than oMigen's.

**Comment by whitequark** [comment]
cc @programmerjake--you mentioned that nMigen elaboration is very slow, and is even slower on PyPy; this seems to be why.

**Comment by whitequark** [comment]
cc @programmerjake --you mentioned that nMigen elaboration is very slow, and is even slower on PyPy; this seems to be why.

**Comment by whitequark** [comment]
Fixed in 4ee82c95.
